### PR TITLE
torch.compile

### PR DESCRIPTION
**Pull request by balakhnov**
## What this does

Explain what this PR does. Feel free to tag your PR with the appropriate label(s).

Examples:
| Title | Label |
|----------------------|-----------------|
| Fixes #[issue] | (🐛 Bug) |
| Adds new dataset | (🗃️ Dataset) |
| Optimizes something | (⚡️ Performance) |

## How it was tested

Explain/show how you tested your changes.

Examples:

- Added `test_something` in `tests/test_stuff.py`.
- Added `new_feature` and checked that training converges with policy X on dataset/environment Y.
- Optimized `some_function`, it now runs X times faster than previously.

## How to checkout & try? (for the reviewer)

Provide a simple way for the reviewer to try out your changes.

Examples:

```bash
pytest -sx tests/test_stuff.py::test_something
```

```bash
lerobot-train --some.option=true
```

## SECTION TO REMOVE BEFORE SUBMITTING YOUR PR

**Note**: Anyone in the community is free to review the PR once the tests have passed. Feel free to tag
members/contributors who may be interested in your PR. Try to avoid tagging more than 3 people.

**Note**: Before submitting this PR, please read the [contributor guideline](https://github.com/huggingface/lerobot/blob/main/CONTRIBUTING.md#submitting-a-pull-request-pr).

### CODE REVIEW

File: configs/smol_and_fast_30000.json [hunk 1]
```diff
@@ -38,7 +38,8 @@
         "drop_n_last_frames": true,
         "grad_clip_norm": 1,
         "relaxed_action_decoding": true,
-        "crop_shape": [84, 84]
+        "crop_shape": [84, 84],
+        "max_input_seq_len": 128
```
Comment (southfreebird) [inline configs/smol_and_fast_30000.json:42]: Is this a real value?

File: src/lerobot/policies/smolandfast/modeling_smolandfast.py [hunk 2]
```diff
@@ -244,6 +244,12 @@ def create_input_tokens(self, states, images, lang_text, actions=None):
 
             max_len = max([len(seq) for seq in prefix_tokens])
 
+            if max_len > self.max_input_seq_len:
+                print("Sequence length is above the maximum sequence length. "
```
Comment (southfreebird) [inline src/lerobot/policies/smolandfast/modeling_smolandfast.py:248]: log.warning?

File: src/lerobot/policies/factory.py [hunk 3]
```diff
@@ -406,6 +406,9 @@ def make_policy(
     policy.to(cfg.device)
     assert isinstance(policy, torch.nn.Module)
 
-    # policy.model.model.vlm.text_model = torch.compile(policy.model.model.vlm.text_model)#, mode="reduce-overhead")
+    if cfg.type == "smolandfast":
+        policy.model.vlm.model.text_model = torch.compile(policy.model.vlm.model.text_model)
+        policy.model.vlm.model.connector = torch.compile(policy.model.vlm.model.connector)
+        policy.model.vlm.model.vision_model = torch.compile(policy.model.vlm.model.vision_model)
```
Comment (southfreebird) [inline src/lerobot/policies/factory.py:412]: tbh I expect something more sophisticated here :)

**Review by southfreebird — APPROVED** [review]
LGTM

**Comment by southfreebird** [comment]
@balakhnov Let's merge it :)

File: src/lerobot/policies/factory.py (hunk 3 above)
Comment (zaringleb) [reply src/lerobot/policies/factory.py:412]: Hmm, I usually saw wrapping for the training stage only 🤔
